contracts/pallet-ibc/src/ics20/memo.rs

```diff
@@ -190,3 +183,21 @@ impl<T: Config + Send + Sync, S: Module + Clone + Default + PartialEq + Eq + Deb
 		self.inner.on_timeout_packet(ctx, output, packet, relayer)
 	}
 }
+
+impl<T: Config + Send + Sync, S: Module + Clone + Default + PartialEq + Eq + Debug> Memo<T, S> {
+	fn process_memo(packet: &mut Packet) -> Result<(), Error> {
+		let packet_data: PacketData =
+			serde_json::from_slice(packet.data.as_slice()).map_err(|e| {
+				Error::implementation_specific(format!("Failed to decode packet data {:?}", e))
+			})?;
+		let receiver = <T as Config>::AccountIdConversion::try_from(packet_data.receiver.clone())
+			.map_err(|_| {
+				Error::implementation_specific(format!("Failed to parse receiver account"))
+			})?
+			.into_account();
+		<T as Config>::HandleMemo::execute_memo(&packet_data, receiver).map_err(|e| {
```

Comment:
this function doesn't return an `ack`?

Reply:
No, we only get it from the inner transfer module